torch/distributed/tensor/_dispatch.py

```diff
@@ -288,15 +288,30 @@ def default_tensor(spec: DTensorSpec) -> torch.Tensor:
         if op_info.schema.is_inplace_op():
             # inplace op should return self instead of re-wrapping
             if output_sharding.output_spec is not None:
+                output_spec = output_sharding.output_spec
+                assert isinstance(output_spec, DTensorSpec)
+                assert isinstance(args[0], dtensor.DTensor)
+
+                # Inplace operations that require redistribution are not supported
+                # because they break aliasing semantics. If there are views into
+                # the tensor, simply overwriting _local_tensor won't update the views.
+                if output_sharding.needs_redistribute:
+                    raise RuntimeError(
+                        f"{op_call}: in-place operations that require redistribution "
+                        f"are not supported. The operation would change placement from "
+                        f"{args[0]._spec.placements} to {output_spec.placements}, "
+                        f"which requires redistribution and breaks aliasing semantics. "
+                        f"Please use the out-of-place version of this operation instead."
+                    )
+
+                # update the spec for all inplace ops to handle placement changes
+                # that don't require redistribution
+                args[0]._spec = output_spec
```

Comment:
I think it's still OK (and in fact preferred) for the squeeze_ logic to live inside the squeeze_ branch. Most inplace ops do NOT change the tensor meta. So because no redistribution could have occurred, the input cannot have a placement change.